### PR TITLE
Adding Cashier Service and Test

### PR DESCRIPTION
**Pull request by DianaMD21**
This branch contains the implementation and interface of the cashier service, as well as its test.
I also changed the type of the Code attribute on the Employee entity. It was of type Long, and I changed it to String. Each employee type will have a different prefix for their code. For example, a cashier will have the prefix "CA" before the number.

### CODE REVIEW

File: src/services/implementation/ClientServiceImpl.java [hunk 1]
```diff
@@ -1,73 +1,11 @@
...
-    UserUtil.fetchUser(client, entity);
-    return Optional.of(client);
-  }
-
-  @Override
-  public List<entities.Client> findAll() {
-    return this.clients.stream()
-        .filter(c -> c.getStatus() != StatusEnum.DELETED)
-        .collect(Collectors.toList());
+    super(0L, Client.class, currentId -> currentId + 1, ClientUtil::fetchClient);
```
Comment (reynoldmorel) [inline src/services/implementation/ClientServiceImpl.java:9]: Within base service util create 2 maps:

1. To map default key with its generator function. Ex: `public static final Map<Class<?>, BiFunction<?, ? , ?>> ENTITY_TO_FETCH_FUNCTION_MAP = Map.of(Client.class, ClientUtil::fetchClient, ....);` 
2. To map the entity class with its fetch function. Ex: `public static final Map<Class<?>, BiFunction<?, ? , ?>> ENTITY_KEY_TO_GENERATOR_FUNCTION_MAP = Map.of(Long.class, currentId -> currentId + 1);`

And then create a functions to get the values of each map respectively given a key.

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/services/implementation/ClientServiceImpl.java (hunk 1 above)
Comment (DianaMD21) [reply src/services/implementation/ClientServiceImpl.java]: What i did is create the map, and BaseServiceImpl simply calls it with .get(Entity.class) and it returns the BiFunction that fetch the correct entity

**Comment by DianaMD21** [comment]
What i did is create the map, and BaseServiceImpl simply calls it with .get(Entity.class) and it returns the BiFunction that fetch the correct entity

File: src/services/implementation/BaseServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  protected K currentId;
+  private final Function<K, K> generateNextId;
+  private final BiFunction<E, E, E> fetchEntity;
+
+  public BaseServiceImpl(K currentId, Class<E> entityClass) {
+    this.entities = new ArrayList<>();
+    this.entityClass = entityClass;
+    this.currentId = currentId;
+    this.generateNextId =
+        (Function<K, K>) BaseEntityUtil.ENTITY_KEY_TO_GENERATOR_FUNCTION_MAP.get(Long.class);
```
Comment (reynoldmorel) [inline src/services/implementation/BaseServiceImpl.java:32]: Please don't do this, what if `K` is String?

File: src/util/BaseEntityUtil.java [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
+import java.util.function.Function;
 
 public final class BaseEntityUtil {
+
+  public static final Map<Class<?>, Function<?, ?>> ENTITY_KEY_TO_GENERATOR_FUNCTION_MAP =
+      Map.of(Long.class, (Long currentId) -> currentId + 1);
+  public static final Map<Class<?>, BiFunction<?, ?, ?>> ENTITY_TO_FETCH_FUNCTION_MAP =
+      Map.of(
+          Client.class,
+          (target, source) -> ClientUtil.fetchClient((Client) target, (Client) source),
```
Comment (reynoldmorel) [inline src/util/BaseEntityUtil.java:18]: maybe ` (Client target, Client source) -> ClientUtil.fetchClient(target, source)` is cleaner?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/util/BaseEntityUtil.java (hunk 3 above)
Comment (DianaMD21) [reply src/util/BaseEntityUtil.java:18]: solved

File: src/util/BaseEntityUtil.java [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
+import entities.Client;
+import java.util.Map;
 import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 
 public final class BaseEntityUtil {
+
+  public static final Map<Class<?>, Function<?, ?>> ENTITY_KEY_TO_GENERATOR_FUNCTION_MAP =
+      Map.of(Long.class, (Long currentId) -> currentId + 1);
```
[the remark this answers is not in the document]
Comment (DianaMD21) [reply src/util/BaseEntityUtil.java:14]: Yes, if I don't specify it, the parameter is interpreted as an Object, and the "+" cannot be applied to it

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**